waspc/src/Wasp/Generator/NpmInstall.hs

```diff
@@ -42,26 +43,34 @@ installNpmDependenciesWithInstallRecord spec dstDir = runExceptT $ do
 
   allNpmDeps <- getAllNpmDeps spec & onLeftThrowError
 
-  liftIO (areThereNpmDepsToInstall allNpmDeps dstDir) >>= \case
-    False -> pure ()
-    True -> do
-      -- In case anything fails during installation that would leave node modules in
-      -- a broken state, we remove the log of installed npm deps before we start npm install.
-      liftIO $ forgetInstalledNpmDepsLog dstDir
+  shouldInstallNpmDeps <-
+    liftIO $
+      or
+        <$> sequence
+          [ -- Users might by accident delete node_modules dir, so we check if it exists
+            -- before assuming that we don't need to install npm deps.
+            not <$> doesNodeModulesDirExist waspProjectDirPath,
```

Comment:
This can still cause problems if they do something more specific (i.e., delete some packages without deleting the entire directory).

@Martinsos What was the reason for not running NPM install every time and letting NPM see whether it needs to install anything? I know we had a reason (a bug, IIRC), but I can't remember.

Reply:
If I understood this correctly, the optimisation was done to prevent `npm install` from running if the deps didn't change. I guess `npm` is smart and optimised enough to figure out quickly that it already has all the deps installed? 

If just run it every time without this optimization, it would be an overhead, maybe negligible but the system would be more robust.

Reply:
Yeah hm I can't remember exactly why we have this check -> I believe it was as you both assumed, which was just to avoid unneccesary runs of `npm install`. Running `npm install` on every file save, it seemed very wasteful, and it potentially makes quite fast iteration cycle quite longer. That said, I can't remember if we tested/measuer these times or just assumed it.

Reply:
I think it was initially implemented by Martijn, but then (during restructuring) it either caused a bug or was necessary to fix it.

I can dig into it if it's important, but probably not, right?